vpr/src/route/route_util.h

```diff
@@ -1,9 +1,19 @@
 #ifndef VPR_ROUTE_UTIL_H
 #define VPR_ROUTE_UTIL_H
 #include "vpr_types.h"
+#include "draw_types.h"
+#include "draw_global.h"
 
 vtr::Matrix<float> calculate_routing_avail(t_rr_type rr_type);
-vtr::Matrix<float> calculate_routing_usage(t_rr_type rr_type, bool is_flat);
+
+/**
+ * @brief Calculates routing usage over entire grid
+ * Collects all in-use nodes and records number of used resources
+ * in each x/y channel. Takes into consideration visible layers
+ * for multi-layered architectures. Also takes into consideration
```

Comment:
I think this comment needs updating.

1. @brief: Calculates and returns the usage over the entire grid for the specified type of rr_node to the usage array. The usage is recorded at each (x,y) location. 

@param rr_type: Type of rr_node that we are calculating the usage of; can be CHANX or CHANY.
@param is_flat: Is the flat router being use or not?
@param only_visible: If true, only record the usage of rr_nodes on layers that are visible according to the current drawing settings.

rr_type: type of rr_node to update usage for (currently supports CHANX or CHANY)